File: 2017 - 2/Programação de Computadores/All Netbeans Projects/NetBeansProjects/C++/Escultura.cpp
#include "Escultura.h"

#include <iostream>
using namespace std;

namespace pc {

    Escultura::Escultura(string titulo, string artista, string material, int ano, double altura) :
    ObraDeArte(titulo, artista, material, ano), altura(altura) {
    }

    Escultura::~Escultura() {
    }

    double Escultura::getAltura() const {
        return altura;
    }

    void Escultura::imprime() {
        cout << "=======================================" << endl;
        cout << "=            ESCULTURA                =" << endl;
        cout << "=======================================" << endl;
        cout << "Título   : " + this->getTitulo() << endl;
        cout << "Artista  : " + this->getArtista() << endl;
        cout << "Material : " + this->getMaterial() << endl;
        cout << "Ano      : ";
        cout << this->getAno() << endl;
        cout << "Altura   : ";
        cout << this->getAltura() << endl;
    }

}

File: 2017 - 2/Programação de Computadores/delano prova/abstract Class/abstract/src/br/ufscar/dc/pc/Escultura.java
package br.ufscar.dc.pc;

/**
 * Classe Escultura
 *
 * @author Delano Medeiros Beder
 */
public class Escultura extends ObraDeArte {

    /*
     * Declaração dos atributos da classe
     */
    private double altura;

    /*
     * Declaração do construtor da classe
     */
    public Escultura(String titulo, String artista, String material, int ano,
            double altura) {
        super(titulo, artista, material, ano);
        this.altura = altura;
    }

    /*
     * Declaração dos métodos da classe
     */
    @Override
    public void imprime() {
        System.out.println("=======================================");
        System.out.println("=            ESCULTURA                =");
        System.out.println("=======================================");
        System.out.println("Título   : " + this.getTítulo());
        System.out.println("Artista  : " + this.getArtista());
        System.out.println("Material : " + this.getMaterial());
        System.out.println("Ano      : " + this.getAno());
        System.out.println("Altura   : " + this.altura);
    }
}
